pkg/reconciler/kubernetes/adapter.go

```diff
@@ -213,6 +213,18 @@ func (g *kubeClientAdapter) newNamespaceUnstruct(namespace string) (*unstructure
 	return nsUnstructs[0], nil
 }
 
+func (g *kubeClientAdapter) DeleteResourceByKindAndNameAndNamespace(kind, name, namespace string) (*Resource, error) {
+	metadata, err := g.kubeClient.DeleteResourceByKindAndNameAndNamespace(
+		kind, name, namespace, metav1.DeleteOptions{})
+	deletedResource := toResource(metadata)
+	if err != nil && !k8serr.IsNotFound(err) {
+		g.logger.Errorf("Failed to delete Kubernetes unstructured resource kind='%s', name='%s', namespace='%s': %s",
+			kind, name, namespace, err)
+		return deletedResource, err
+	}
+	return deletedResource, nil
+}
```

Comment:
If this code is not needed by other component reconcilers, then we can move it inside the Eventing reconciler only.